packages/good-design/src/advanced/web3action/Web3Action.tsx

```diff
@@ -99,7 +99,7 @@ export const Web3ActionButton: FC<Web3ActionProps> = withTheme({
   // account/chainId changes and re-try to perform current step action
   useEffect(() => {
     const continueSteps = async () => {
-      if (!account) {
+      if (!account && !chainId) {
```

Comment:
what happens if  there's no account connected? dont we need to ask user to connnect one?